### DISCUSSION TITLE
Flash Wifi Example/ wifi streaming through gap8

### DISCUSSION THREAD

**Discussion by crazyflie-aljanabi** [bitcraze/discussions#595] [Q&A] [answered]
Hi,

i did Clone, build and flash the bootloader with an Olimex ARM-USB-TINY-H JTAG using the following commands, correctly:

$ git clone https://github.com/bitcraze/aideck-gap8-bootloader.git
$ cd aideck-gap8-bootloader
$ docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/ftdi/olimex-arm-usb-tiny-h.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make all image flash'

But i get on the console only the following: Reset to bootloader mode ...
Could not save cache, no writable directory
Could not save cache, no writable directory
Skipping bcAI:esp, not in the target list
Deck bcAI:gap8

I repeated the Gap8 bootloader many times but no change?

Another question: i want make the wifi Flash but i don't understand how i do the following: In your wifi list you should see ‘WiFi streaming example’. If you do, please connect to it.

Clone the [AIdeck example repository](https://github.com/bitcraze/aideck-gap8-examples)

Then run:

cd examples/other/wifi-img-streamer
python opencv-viewer.py

I don't have the wifi-img-streamer directory, how can i do it please?

Thank you for your help!

**Comment by hmllr** [comment]
Hi :)
There seem to be multiple approaches for flashing mixed up a bit in your text, so I am not completely sure if I understand what the problem is - correct me if I got it wrong! 

First, as you correctly wrote, you need to flash the bootloader with a debugger as the olimex. Then you need to disconnect this cable (and power cycle the drone) and switch to the cfclient and update all firmwares from there (choose the newest release). Afterwards I would again power cycle (turn off and on the drone) and then wait for the wifi network to be established and connect to it with your computer.

The viewer python script is in the directory to which you posted a link, https://github.com/bitcraze/aideck-gap8-examples/blob/master/examples/other/wifi-img-streamer/opencv-viewer.py .

Hope this helps,
Hanna

**Comment by crazyflie-aljanabi** [comment]
Dear Hanna,

I am writing to correct the information I previously provided. Here are the details:

Bootloader: I successfully flashed the Gap8 bootloader onto the GAP8 using the Olimex ARM-USB-TINY-H JTAG. I did not encounter any issues with this step.

Flashing Wifi Example: For this step, I downloaded the "aideck_gap8_wifi_img_streamer_with_ap.bin" file and executed the following command: "cfloader flash aideck_gap8_wifi_img_streamer_with_ap.bin deck-bcAI:gap8-fw -w radio://0/80/2M/E7E7E7E7E7". However, when I checked the console, I only saw the following message: 
"Reset to bootloader mode ... 
Could not save cache, no writable directory
Could not save cache, no writable directory
Skipping bcAI:esp,"

 **Here is what is missing:**

, not in the target list. Deck bcAI:gap8
reset to bootloader
| 0% Writing to bcAI:gap8 deck memory
/ 1% Writing to bcAI:gap8 deck memory
...
\ 99% Writing to bcAI:gap8 deck memory
| 100% Writing to bcAI:gap8 deck memory

Another question I have is how to execute the following command: "cd examples/other/wifi-img-streamer python opencv-viewer.py"? I could not find a directory with the name "wifi-img-streamer" on my Ubuntu VM.

Have i to Setup development environment (Setting up docker and the autotiler)?


Thank you for your attention to this matter.

Best regards,
Ali

**Reply by hmllr** [reply]
Hi Ali,

This looks like the flashing is working correctly actually :)

You need to follow the instructions in the getting started guide: https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/

To find the viewer script you need to clone the ai-deck examples directory, with 
`git clone git@github.com:bitcraze/aideck-gap8-examples.git` if you have an ssh key or with `git clone https://github.com/bitcraze/aideck-gap8-examples.git` otherwise. In there you'll find the file under "examples/other/wifi-img-streamer/opencv-viewer.py".

To only receive/view the images you don't need docker or the autotiler, you only need this if you want to complie your own code.

I hope this helps,
Hanna

**Comment by crazyflie-aljanabi** [comment]
Hi,

I tried to flash the wifi_img_streamer_with_ap.bin, but it hangs at 6%... i just tried building flashing the gap8 bootloader again with my Olimex ARM-USB-TINY-H JTAG, but it tell me that there is no space (maybe the memory of my AI-deck full?). Oherwise i did this command git clone https://github.com/bitcraze/aideck-gap8-examples.git , but i couldn't find the file "examples/other/wifi-img-streamer python opencv-viewer.py". I have AI-deck 1.1 with the revision rev. C . 

I updated Crazyflie and AIdeck firmware many times but i didn't see  in the console tab : **ESP32: I (910) SYS: Initialized** after connecting to the crazyflie!

What can i doing please?


**Note:** Important to understand.I want the Crazyflie to be able to recognize objects during flight using its camera and transmit video to my computer. The Crazyflie must avoid objects by processing the camera feed and flying accordingly.

Thank you for your attention to this matter.

Best regards,
Ali

**Reply by hmllr** [reply]
Can you try to update all the microcontrollers to the newest firmware again? Usually this kind of behavior happens when the ESP firmware is not up to date.  You can follow the https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/ getting started guide. Make sure to only have the AI-deck (and no other deck, so now flow deck or anything else) connected while doing it, because this can cause issues with the update (for unknown reasons).

For the python script, I had a typo in the path (as I copied it from your question, my bad, sorry for that) the correct path is: "examples/other/wifi-img-streamer/opencv-viewer.py" of course this file is inside the aideck-gap8-examples directory.

Hope this helps,
Hanna

**Comment by crazyflie-aljanabi** [comment, marked as answer]
Hi,


I have successfully performed the 'Flash Wifi Example' and I can now see the image from the AI-deck camera on my computer!

Thank you very much for your help!


I have now another topic: I would like to implement and test object detection for my crazyflie quadcopter on the host computer. The crazyflie must be capable of performing automated flight functions based on camera-based object detection. How can I do this? I need your help please on this topic. What should I do and which code should I use for this?

Best regards

Ali

________________________________
Von: hmllr ***@***.***>
Gesendet: Montag, 6. März 2023 11:38:10
An: bitcraze/discussions
Cc: Al-Janabi, Ali Salman Aboob; Author
Betreff: Re: [bitcraze/discussions] Flash Wifi Example/ wifi streaming through gap8 (Discussion #595)


Can you try to update all the microcontrollers to the newest firmware again? Usually this kind of behavior happens when the ESP firmware is not up to date. You can follow the https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/ getting started guide. Make sure to only have the AI-deck (and no other deck, so now flow deck or anything else) connected while doing it, because this can cause issues with the update (for unknown reasons).

For the python script, I had a typo in the path (as I copied it from your question, my bad, sorry for that) the correct path is: "examples/other/wifi-img-streamer/opencv-viewer.py" of course this file is inside the aideck-gap8-examples directory.

Hope this helps,
Hanna

**Reply by hmllr** [reply]
Happy to hear that it is working!

However, please open a new question if the question is not related to the title anymore, otherwise people will have a hard time finding it.

Then, best you formulate your question as specific as possible (like "how do I send data from the gap8 to an application on the Crazyflie") and check first if the question already exists with a quick search.

**Comment by crazyflie-aljanabi** [comment]
ok, i opened a new question and formulate my question as follow: ,,how do I send data from the gap8 to an application on the Crazyflie- objects detection"


thank you very much!


Best regards

Ali



________________________________
Von: hmllr ***@***.***>
Gesendet: Mittwoch, 8. März 2023 09:08:29
An: bitcraze/discussions
Cc: Al-Janabi, Ali Salman Aboob; Author
Betreff: Re: [bitcraze/discussions] Flash Wifi Example/ wifi streaming through gap8 (Discussion #595)


Happy to hear that it is working!

However, please open a new question if the question is not related to the title anymore, otherwise people will have a hard time finding it.

Then, best you formulate your question as specific as possible (like "how do I send data from the gap8 to an application on the Crazyflie") and check first if the question already exists with a quick search.
